test/e2e/create_test.go

```diff
@@ -376,6 +400,40 @@ func (tc *testContext) waitForWindowsNodes(nodeCount int32, expectError, checkVe
 	return err
 }
 
+// usernameFromNode identifies the username associated with a given node by comparing addresses
+func (tc *testContext) usernameFromNode(node *v1.Node) (string, error) {
+	if node == nil {
+		return "", errors.New("cannot get username for nil node")
+	}
+
+	nodes, err := tc.client.K8s.CoreV1().Nodes().List(context.TODO(),
+		metav1.ListOptions{LabelSelector: v1.LabelOSStable + "=windows"})
+	if err != nil {
+		return "", errors.Wrap(err, "unable to list nodes")
+	}
+	instanceConfigMap, err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(),
+		controllers.InstanceConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get instance configmap")
+	}
+
+	// Get list of instances that are expected to be Nodes from ConfigMap
+	byohInstances, err := instances.ParseInstances(instanceConfigMap.Data, nodes)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to read instances from configmap %s", instanceConfigMap.Name)
+	}
+	for _, instance := range byohInstances {
+		nodeAddressStr, err := controllers.GetAddress(node.Status.Addresses)
+		if err != nil {
+			return "", errors.Errorf("unable to find address for node %s", node.Name)
+		}
+		if instance.Address == nodeAddressStr {
+			return instance.Username, nil
+		}
+	}
+	return "", errors.Errorf("unable to find Windows instance associated with node %s", node.Name)
+}
```

Comment:
I dont think this function is necessary at all. the test suites knows what the username is as it creates it.
We supply the expected value so we know what it should be, the username should be read directly from the configmap, instead of calling `ParseInstances`. You already have the node so you know the ConfigMap entry key

Reply:
You're right, thanks for pointing this out.